js/src/components/ads-account-select-control/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import AppSelectControl from '.~/components/app-select-control';
+import useExistingGoogleAdsAccounts from '.~/hooks/useExistingGoogleAdsAccounts';
+
+/**
+ * @param {Object} props The component props
+ * @param {string} [props.value] The selected value. IF no value is defined, then the first option is selected and onChange function is triggered.
+ * @param {Function} [props.onChange] Callback when the select value changes.
+ * @return {JSX.Element} An enhanced AppSelectControl component.
+ */
+const AdsAccountSelectControl = ( {
+	value,
+	onChange = () => {},
+	...props
+} ) => {
+	const { existingAccounts: accounts = [] } = useExistingGoogleAdsAccounts();
+
+	const options = accounts.map( ( acc ) => ( {
+		value: acc.id,
+		label: `${ acc.name } (${ acc.id })`,
+	} ) );
+
+	useEffect( () => {
+		// Triggers the onChange event in order to pre-select the initial value
+		if ( value === undefined ) {
+			onChange( options[ 0 ]?.value );
+		}
+	}, [ options, onChange, value ] );
```

Comment:
Although the current use of this component won't have issues in the runtime, from the perspective of component design, I'm concerned about its potential issue of infinite rendering loop.

```jsx
function TestComponent() {
	const [ _, setValue ] = useState( 0 );

	return (
		<AdsAccountSelectControl
			onChange={ () => {
				// Update some states that always trigger a top-down re-rendering
				setValue( ( v ) => v + 1 );
			} }
		/>
	);
}
```

Reply:
@eason9487 The way we have it, we'll always need to pass on the value prop which should not cause the infinite rendering.